src/backend/marsha/markdown/api.py

```diff
@@ -0,0 +1,159 @@
+"""Declare API endpoints with Django RestFramework viewsets."""
+
+import uuid
+
+from django.urls import reverse
+
+from rest_framework import mixins, viewsets
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST
+
+from marsha.core import permissions as core_permissions
+from marsha.core.api import ObjectPkMixin
+from marsha.core.utils.url_utils import build_absolute_uri_behind_proxy
+
+from . import serializers
+from .models import MarkdownDocument
+from .utils.converter import LatexConversionException, render_latex_to_image
+
+
+class MarkdownDocumentViewSet(
+    ObjectPkMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Viewset for the API of the Meeting object."""
+
+    queryset = MarkdownDocument.objects.prefetch_related("translations").all()
+    serializer_class = serializers.MarkdownDocumentSerializer
+
+    permission_classes = [
+        core_permissions.IsTokenResourceRouteObject
+        & (core_permissions.IsTokenInstructor | core_permissions.IsTokenAdmin)
+    ]
+
+    @action(
+        methods=["get"],
+        detail=False,
+        url_path="lti-select",
+        permission_classes=[core_permissions.HasPlaylistToken],
+    )
+    def lti_select(self, request):
+        """Get selectable content for LTI.
+
+        Calling the endpoint returns a new uuid for a markdown
+        document and a list of available documents.
+
+        Parameters
+        ----------
+        request : Type[django.http.request.HttpRequest]
+            The request on the API endpoint
+
+        Returns
+        -------
+        Type[rest_framework.response.Response]
+            HttpResponse carrying selectable content as a JSON object.
+
+        """
+        new_uuid = str(uuid.uuid4())
+        new_url = build_absolute_uri_behind_proxy(
+            self.request,
+            reverse("markdown:markdown_document_lti_view", args=[new_uuid]),
+        )
+
+        markdown_documents = serializers.MarkdownDocumentSelectLTISerializer(
+            MarkdownDocument.objects.filter(
+                playlist__id=request.user.token.payload.get("playlist_id")
+            ),
+            many=True,
+            context={"request": self.request},
+        ).data
+
+        return Response(
+            {
+                "new_url": new_url,
+                "markdown_documents": markdown_documents,
+            }
+        )
+
+    @action(
+        methods=["post"],
+        detail=True,  # Add the document ID, not used for now
```

Comment:
Yes, I was wondering all along whether talking about `document` which is simpler than `markdown document` was ok in the `markdown` modules, but you are right maybe it would be more explicit to specify it everywhere (in case someone wants to grep).